cfgmgr/nbrmgr.cpp

```diff
@@ -294,8 +309,253 @@ void NbrMgr::doTask(Consumer &consumer)
+            {
+                SWSS_LOG_ERROR("Route entry on dev %s delete failed for '%s'", nbr_odev.c_str(), kfvKey(t).c_str());
+            }
+            else
+            {
+                SWSS_LOG_NOTICE("Route entry on dev %s deleted for '%s'", nbr_odev.c_str(), kfvKey(t).c_str());
+            }
+
+            if (!delKernelNeigh(nbr_odev, ip_address))
+            {
+                SWSS_LOG_ERROR("Neigh entry on dev %s delete failed for '%s'", nbr_odev.c_str(), kfvKey(t).c_str());
+            }
+            else
+            {
+                SWSS_LOG_NOTICE("Neigh entry on dev %s deleted for '%s'", nbr_odev.c_str(), kfvKey(t).c_str());
+            }
+            SWSS_LOG_DEBUG("Deleted voq neighbor %s from kernel", kfvKey(t).c_str());
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+bool NbrMgr::getVoqInbandInterfaceName(string &ibif)
+{
+
+    vector<string> keys;
+    m_cfgVoqInbandInterfaceTable->getKeys(keys);
+
+    if (keys.empty())
+    {
+        SWSS_LOG_NOTICE("Voq Inband interface is not configured!");
+        return false;
+    }
+    //key:"alias" = inband interface name
+    vector<string> if_keys = tokenize(keys[0], config_db_key_delimiter);
+    ibif = if_keys[0];
+    return true;
+}
+
+bool NbrMgr::addKernelRoute(string odev, IpAddress ip_addr)
+{
+    string cmd, res;
+
+    SWSS_LOG_ENTER();
+
+    string ip_str = ip_addr.to_string();
+
+    if(ip_addr.isV4())
+    {
+        cmd = string("") + IP_CMD + " route add " + ip_str + "/32 dev " + odev;
+        SWSS_LOG_NOTICE("IPv4 Route Add cmd: %s",cmd.c_str());
+    }
+    else
+    {
+        cmd = string("") + IP_CMD + " -6 route add " + ip_str + "/128 dev " + odev;
+        SWSS_LOG_NOTICE("IPv6 Route Add cmd: %s",cmd.c_str());
+    }
+
+    int32_t ret = swss::exec(cmd, res);
+
+    if(ret)
+    {
+        /* Just log error and return */
+        SWSS_LOG_ERROR("Failed to add route for %s, error: %d", ip_str.c_str(), ret);
+        return false;
+    }
+
+    SWSS_LOG_INFO("Added route for %s on device %s", ip_str.c_str(), odev.c_str());
+    return true;
+}
+
+bool NbrMgr::delKernelRoute(IpAddress ip_addr)
+{
+    string cmd, res;
+
+    SWSS_LOG_ENTER();
+
+    string ip_str = ip_addr.to_string();
+
+    if(ip_addr.isV4())
+    {
+        cmd = string("") + IP_CMD + " route del " + ip_str + "/32";
+        SWSS_LOG_NOTICE("IPv4 Route Del cmd: %s",cmd.c_str());
+    }
+    else
+    {
+        cmd = string("") + IP_CMD + " -6 route del " + ip_str + "/128";
+        SWSS_LOG_NOTICE("IPv6 Route Del cmd: %s",cmd.c_str());
+    }
+
+    int32_t ret = swss::exec(cmd, res);
+
+    if(ret)
+    {
+        /* Just log error and return */
+        SWSS_LOG_ERROR("Failed to delete route for %s, error: %d", ip_str.c_str(), ret);
+        return false;
+    }
+
+    SWSS_LOG_INFO("Deleted route for %s", ip_str.c_str());
+    return true;
+}
+
+bool NbrMgr::addKernelNeigh(string odev, IpAddress ip_addr, MacAddress mac_addr)
```

Comment:
Just curious, why don't we use `setNeighbor ` instead?

Reply:
There is no specific reason why setNeighbor() is not used. While looking for kernel entry programming, I felt using shell commands to add/del kernel entries seemed simple. So I added this function and other functions for delete neigh, static route add and static route delete functions that use the shell commands for kernel entry programming.